nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -33,38 +32,21 @@ class ONNXGraph:
 
     def __init__(self, onnx_model: onnx.ModelProto):
```

Comment:
Is it possible to make all methods inside the `ONNXGraph` static (i.e. add `@staticmethod`)?

Reply:
We've just discussed this with @kshpv. Having all methods in a class as static really makes you consider whether you need a class at all. I suppose that here we do need a class, namely - its objects as entities with independent states and lifetimes.

Reply:
@kshpv Is there at least one method inside the `ONNXGraph` that modifies the state of the object (i.e. `self`)?

As far as I can see this class looks like a collection of independent methods (utilities) for working with `onnx.ModelProto`.

Reply:
> @kshpv Is there at least one method inside the `ONNXGraph` that modifies the state of the object (i.e. `self`)?
> 
> As far as I can see this class looks like a collection of independent methods (utilities) for working with `onnx.ModelProto`.

ONNXGraph provides the methods to get the necessary infromation from the ONNX model. It doesn't change it. 
To apply transformation on the ONNX model, we have ONNXModelTransformer.

Reply:
So, I still have my question. Why we can't decorate all methods inside the ONNXGraph as `@staticmethod`? Or better, remove this class and create the module with all methods that regard the `onnx.ModelProto`. For example, `nncf/experimental/onnx/graph/utils.py`

Reply:
Your approach is possible, I implemented it as the first iteration. 
The problem here is that the developer should control the correct statement of the ONNX model during quantization, meaning to do infer_shape while it is necessary. 
I am working on another one, where ONNXGraph stores the model and does infer shape only when it is necessary. In that case, the developer doesn't have to care about the correct statement of ONNX model.

Reply:
I've pushed the new version. @andrey-churkin, please, take a look

Reply:
@vshampor, please also take a look

Reply:
I've looked as well and suggest merging this PR.